### ISSUE TITLE
Not Working On First Request

### ISSUE THREAD

**Issue by ambrizals** [Morgbn/nuxt-csurf#54]
## Project Information :
- Nuxt Version : ^3.9.0
- Library Version (nuxt-csurf) : ^1.6.5
- NodeJS : v18.20.3
- Mode : SSR

Hello i got some issue here, i just realise API is not accessible at first request and nuxt throw some error on CLI.

```
WARN  [Vue Router warn]: No match found for location with path "/api/trpc/v2.dashboard.topDaily"
WARN  [Vue Router warn]: No match found for location with path "/api/trpc/v2.dashboard.topDaily"
WARN  [Vue Router warn]: No match found for location with path "/api/trpc/v2.dashboard.table"
```

After first request and try to navigate to another page API will working fine, but if we try to remove cache and site data API will not work and we need to reload page to make API be accessible.

Basically all my API was handled by tRPC and i have to modify tRPC client to put CSRF token in header.

```typescript
  // CSRF was only come on client side, if API was hit on server side you need to take token from event context
  const { csrf } = useCsrf()

  const client = createTRPCNuxtClient<AppRouter>({
    links: [
      httpLink({
        url: '/api/trpc',
        headers: {
          ...headers,
          'csrf-token': csrf ?? ctx.ssrContext?.event.context.csrfToken,
        },
        fetch: customFetch as FetchEsque,
      }),
    ],
  })
```

After make some investigation actually API was returning 302 status code and showing CSRF Token Missmatch.

```typescript
async function customFetch(input: RequestInfo, init?: RequestInit & { method: 'GET' }) {
  const { clearSession } = useSessionStore()

  return globalThis.$fetch
    .raw(input.toString(), init)
    .catch(async e => {
      console.log('trpc error client', e)
      if (isFetchError(e) && e.response) {
        if (e.statusCode === 401 && import.meta.client) {
          clearSession({ force: true })
        }
        return e.response
      }
      throw e
    })
    .then(async response => {
      console.log('client error', response._data, response.status, response.statusText)
      // client error <!DOCTYPE html><html><head><meta http-equiv="refresh" content="0; url=/auth/login"></head></html> 302 CSRF Token Mismatch

      return {
        ...response,
        headers: response.headers,
        json: () => Promise.resolve(response._data),
      }
    })
}
```

It is a bug or i have do something wrong in my tRPC client ?
We can check it on my video, in incognito mode at first request data wasn't appear but if i try to refresh API is working.

Thanks.

https://github.com/user-attachments/assets/7402ce47-7f1e-4fb7-9c95-8717ad631225

**Comment by Admiral-Enigma** [comment]
Are you sure this is an CSRF issue? The warnings are from nuxt saying it can't find a trpc api route. A CSRF issue would throw a 403 error.

Do you have the following routing structure for the trpc handler?
![image](https://github.com/user-attachments/assets/87a28745-0bde-463b-88fb-6730787411d8)

**Comment by ambrizals** [comment]
> Are you sure this is an CSRF issue? The warnings are from nuxt saying it can't find a trpc api route. A CSRF issue would throw a 403 error.
> 
> Do you have the following routing structure for the trpc handler? ![image](https://private-user-images.githubusercontent.com/13486531/386861407-87a28745-0bde-463b-88fb-6730787411d8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE4Mzc0NDYsIm5iZiI6MTczMTgzNzE0NiwicGF0aCI6Ii8xMzQ4NjUzMS8zODY4NjE0MDctODdhMjg3NDUtMGJkZS00NjNiLTg4ZmItNjczMDc4NzQxMWQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE3VDA5NTIyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxMGI4OTlmN2FmMGM3MTA2MjQyYTEyZDI2OWVlODg0OWU5ZGYzYjZkZDU0YmViMmQ4MmE5YzEzOTdiODNiYjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4Q30TBAqgT1soo4YCRp3qXpJQEPtoZoNiQWyQTxjJoM)

Yes i follow that instruction but it's not about tRPC setup is about csurf, my project was running well before setup CSRF you can check in my video.

**Comment by fahmifitu** [comment]
@ambrizals Did you solve this issue?

**Comment by ambrizals** [comment]
> [@ambrizals](https://github.com/ambrizals) Did you solve this issue?

Not yet, issue still exists.

**Comment by dl-ias** [comment]
@Morgbn  Any plans when it would be solved?

**Comment by dl-ias** [comment]
For everyone that face same issue, that might be related to `prerendering: true` for specific routes, and/or route caching ([Hybrid rendering](https://nuxt.com/docs/guide/concepts/rendering#hybrid-rendering)) https://github.com/Morgbn/nuxt-csurf/issues/42
In my case disabling `swr` solved issue, as it was enabled to home page:

![Image](https://github.com/user-attachments/assets/76e31023-2407-4a2c-968a-6b7cd27f8635)
